src/storage-client/src/controller/rehydration.rs

```diff
@@ -414,13 +413,41 @@ where
                 }
             }
             StorageCommand::AllowCompaction(frontiers) => {
+                // Remember for rehydration!
+                self.sinces.extend(frontiers.iter().cloned());
+
                 for (id, frontier) in frontiers {
-                    match self.sinks.get_mut(id) {
-                        Some(export) => {
-                            export.description.as_of.downgrade(frontier);
+                    if id.is_user() {
+                        tracing::info!("in rehydration, for {id}: {:?}", frontier);
+                    }
+
+                    if let Some(export) = self.sinks.get_mut(id) {
+                        export.description.as_of.downgrade(frontier);
```

Comment:
I think all of this logic should be moved to the reconciliation function of storage state. This intermediate client is only responsible for recording and replaying the command stream when a client reconnects. Interpretation of the command stream should be left to the final destination.

There we'll have to opportunity to observe the `CreateSource` and `AllowCompaction({})` *at the same time*, because they will be punctuated by the `InitializationComplete` command